delta/sdk/src/main/scala/ch/epfl/bluebrain/nexus/delta/sdk/identities/IdentitiesImpl.scala

```diff
@@ -61,6 +62,14 @@ class IdentitiesImpl private[identities] (
       )
     }
 
+    def fetchRealm(parsedToken: ParsedToken): IO[Realm] =
+      realm
+        .getOrElseUpdate(parsedToken.rawToken, findActiveRealm(parsedToken.issuer))
+        .flatMap {
+          case Some(realm) => IO.pure(realm)
```

Comment:
`IO.fromOption` / `IO.raiseWhen` ?

Reply:
Ended up using OptionT, what do you think?